src/react/databrowser/objects/ObjectDelete.tsx

```diff
@@ -130,6 +129,170 @@ function ConfirmationInput({ toggledFiles }: { toggledFiles: unknown[] }) {
+        <Banner icon={<Icon name="Exclamation-triangle" />} variant="warning">
+          At least one object you want to delete is under Object-Lock retention{' '}
+          <Icon name="Lock" color="buttonSecondary" /> with compliance mode.{' '}
+          <br />
+          Object versions under compliance retention cannot be deleted.
+        </Banner>
+      </Box>
+    );
+  }
+
+  if (warningType === WarningTypes.LEGAL_HOLD) {
+    return (
+      <Box marginTop={spacing.sp8} marginBottom={spacing.sp8}>
+        <Banner icon={<Icon name="Exclamation-triangle" />} variant="warning">
+          At least one object you want to delete is under Legal Hold{' '}
+          <Icon name="Lock" color="buttonSecondary" /> <br />
+          Object versions with Legal Hold cannot be deleted.
+        </Banner>
+      </Box>
+    );
+  }
+
+  return <></>;
+}
+
+export const getMessagesAndRequiredActions = ({
+  numberOfObjects,
+  selectedObjectsAreSpecificVersions,
+  isBucketVersioned,
+  objectsLockedInComplianceModeLength,
+  objectsLockedInGovernanceModeLength,
+  objectsLockedInLegalHoldLength,
+}: {
+  numberOfObjects: number;
+  selectedObjectsAreSpecificVersions: boolean;
+  isBucketVersioned: boolean;
+  objectsLockedInComplianceModeLength: number;
+  objectsLockedInGovernanceModeLength: number;
+  objectsLockedInLegalHoldLength: number;
+}) => {
+  if (isBucketVersioned) {
+    // default version handling
+    if (!selectedObjectsAreSpecificVersions) {
+      return {
+        info: `Delete ${maybePluralize(
+          numberOfObjects,
+          'marker',
+          's',
+          false,
+        )} will be added to the ${maybePluralize(
+          numberOfObjects,
+          'object',
+          's',
+          false,
+        )}.`,
+        warnings: [],
+        checkboxRequired: false,
+        confirmationRequired: false,
+        isDeletionPossible: true,
+      };
+    }
+    // specific version handling
+    else {
+      // can't delete, return early
+      if (
+        objectsLockedInComplianceModeLength > 0 ||
+        objectsLockedInLegalHoldLength > 0
+      ) {
+        const warnings = [];
+
+        if (objectsLockedInComplianceModeLength > 0) {
+          warnings.push(WarningTypes.COMPLIANCE);
+        }
+        if (objectsLockedInLegalHoldLength > 0) {
+          warnings.push(WarningTypes.LEGAL_HOLD);
+        }
+
+        return {
+          info: '',
+          warnings,
+          checkboxRequired: false,
+          confirmationRequired: false,
+          isDeletionPossible: false,
+        };
+      }
+
+      // can delete with proper access
+      if (objectsLockedInGovernanceModeLength > 0) {
+        return {
+          info: `Delete ${maybePluralize(
+            numberOfObjects,
+            'marker',
+            's',
+            false,
+          )} will be added to the ${maybePluralize(
+            numberOfObjects,
+            'object',
+            's',
+            false,
+          )}.`,
```

Comment:
IIUC, here we are trying to delete specific versions, so we shall not have the `delete marker` info